### ISSUE TITLE
Insight Editor Issue

### ISSUE THREAD

**Issue by BeauNouvelle** [TelemetryDeck/Viewer#213]
I'm having an issue while editing an insiqht.

Insight ID: B19EAE98-CE3E-4F62-BE4C-8CE7E5120395

Please describe the issue

Just looking for a way to group models, or platforms based on the device itself.
For example, I can't find a way to separate iOS vs macOS vs iPadOS, etc
Or, iPhone vs iPad vs Mac, etc.

Doing it by ModelName is too granular, whereas iOS isn't granular enough. Thanks.

**Comment by BeauNouvelle** [comment]
Oh in addition to this "Operating System" groups iOS apps running on macOS as iOS.

**Comment by winsmith** [comment]
Hi, thanks for submitting this issue. 

## iPhone vs iPad, easy

One way I can think of regarding grouping signals by platform is, this:

1. Create a "Daily Active Users" insight, call it "iPhone Users" 
2. Add a new filter of type "regex and filter `modelName` for `iPhone(.*)` 
3. Create a second new "Daily Active Users" insight and call it "iPad Users"
4. Add a new filter of type "regex and filter for `iPad(.*)`

Not ideal, but it might get you the information that you need.

## iPhone vs iPad, hard

A more complicated method of comparing iPhone vs iPad vs Mac numbers would be to to convert your insight to a **Custom Query** and then using the [GroupBy Query Type](https://telemetrydeck.com/docs/tql/groupBy/) with a [Regex Extraction Function](https://telemetrydeck.com/docs/tql/dimensionSpec/).

This is a bit more handiwork, but I built this using the playground right now and it seems to work:

```json
{
  "aggregations": [
    {
      "fieldName": "count",
      "name": "count",
      "type": "longSum"
    }
  ],
  "dataSource": "telemetry-signals",
  "dimensions": [
    {
      "type": "extraction",
      "dimension": "modelName",
      "outputName": "shortModelName",
      "extractionFn": {
        "type": "regex",
        "expr": "(^[a-zA-Z]+)(.*)",
        "index": 1,
        "replaceMissingValue": true,
        "replaceMissingValueWith": "unknown"
      }
    }
  ],
  "baseFilters": "thisApp",
  "appID": "A7B2A8ED-9F9F-4839-BC50-E393CD88265C",
  "granularity": "all",
  "queryType": "groupBy"
}
```

## Operating System treats iOS apps running on macOS as iOS

We've heard this from a few users, and we're trying to [account for that in the Swift SDK code](https://github.com/TelemetryDeck/SwiftClient/blob/main/Sources/TelemetryClient/Signal.swift#L165). If the heuristics there don't work for you, let's go bug hunting together 😅 can you pinpoint the macOS version where this is happening?

**Comment by BeauNouvelle** [comment]
This is excellent thanks for that!I’ll try and let you know more on the other thing tomorrow!On 21 Feb 2023, at 7:42 pm, Daniel Jilg ***@***.***> wrote:﻿
Closed #213 as completed.

**Comment by alexstaravoitau** [comment]
The **Custom Query** insight by @winsmith updated to show the iPhone vs iPad breakdown as a donut chart:

```json
{
  "aggregations": [
    {
      "fieldName": "clientUser",
      "name": "count",
      "type": "thetaSketch"
    }
  ],
  "appID": "<app_id>",
  "baseFilters": "thisApp",
  "dimension": {
    "dimension": "modelName",
    "extractionFn": {
      "expr": "(^[a-zA-Z]+)(.*)",
      "index": 1,
      "replaceMissingValue": true,
      "replaceMissingValueWith": "Unknown",
      "type": "regex"
    },
    "outputName": "Device Type",
    "type": "extraction"
  },
  "granularity": "all",
  "metric": {
    "metric": "count",
    "type": "numeric"
  },
  "queryType": "topN",
  "threshold": 200
}
```
